autoload/ale/fixers/clangtidy.vim

```diff
@@ -0,0 +1,48 @@
+scriptencoding utf-8
+" Author: ObserverOfTime <chronobserver@disroot.org>
+" Description: Fixing C/C++ files with clang-tidy.
+
+function! s:set_variables() abort
+    let l:use_global = get(g:, 'ale_use_global_executables', 0)
+    for l:ft in ['c', 'cpp']
+        call ale#Set(l:ft . '_clangtidy_executable', 'clang-tidy')
+        call ale#Set(l:ft . '_clangtidy_use_global', l:use_global)
+        call ale#Set(l:ft . '_clangtidy_checks', [])
+        call ale#Set(l:ft . '_clangtidy_options', '')
+        call ale#Set(l:ft . '_clangtidy_extra_options', '')
+        call ale#Set(l:ft . '_clangtidy_fix_errors', 1)
+        call ale#Set(l:ft . '_build_dir', '')
+    endfor
+endfunction
+
+call s:set_variables()
+
+function! ale#fixers#clangtidy#Var(buffer, name) abort
+    let l:ft = getbufvar(a:buffer, '&filetype')
```

Comment:
The filetype could be a compound of `'c'` and something else. I recommend instead using `'cpp'` if the filetype contains `'cpp'`, and `'c'` otherwise.